### PR TITLE
Clear the tangential overlap after load balancing in DEM

### PR DESCRIPTION
**Pull request by acdaigneault**
# Description of the problem

- Tangential overlap and relative velocity stored values in contact info object were lost during load balancing only for contacts of particles that is moved on other processor.
- Disabling contacts algorithm had a small glitch after load balancing because the identification of the mobility status was not executed at the right place (before exchange of ghost).

# Description of the solution
- Force the resetting of the tangential overlap and relative velocity values during the update of the iterator since it loops on the map of the contact info object. It used the `load_balance_step` boolean to trigger this resetting.
- Move the identification of the mobility after the exchange of ghost particles

### CODE REVIEW

**Comment by blaisb** [comment]
@acdaigneault some tests don't seem to work anymore?

File: include/dem/update_local_particle_containers.h [hunk 1]
```diff
@@ -63,7 +63,8 @@ void
 update_contact_container_iterators(
   pairs_structure &pairs_in_contact,
   typename DEM::dem_data_structures<dim>::particle_index_iterator_map
-    &particle_container);
+    &        particle_container,
+  const bool load_balance_step = false);
```
Comment (blaisb) [inline include/dem/update_local_particle_containers.h:67]: Is load balance_step the right variable name here or should it be "clear_contact_structures" In essence, the fact that we clear when we load balance is one thing, but the action being asked of the structure is to clear it.

File: source/dem/update_local_particle_containers.cc [hunk 2]
```diff
@@ -83,6 +84,14 @@ update_contact_container_iterators(
                   unsigned int particle_two_id = adjacent_map_iterator->first;
                   adjacent_map_iterator->second.particle_two =
                     particle_container[particle_two_id];
+
+                  if (load_balance_step)
```
Comment (blaisb) [inline source/dem/update_local_particle_containers.cc:88]: I would definitely call that variable "clear"

**Review by blaisb — CHANGES_REQUESTED** [review]
Tiny naming changes, nothing significant.

**Comment by blaisb** [comment]
Awesome, will merge after CI

File: source/dem/update_local_particle_containers.cc [hunk 3]
```diff
@@ -83,6 +84,14 @@ update_contact_container_iterators(
                   unsigned int particle_two_id = adjacent_map_iterator->first;
                   adjacent_map_iterator->second.particle_two =
                     particle_container[particle_two_id];
+
+                  if (clear_contact_structures)
+                    {
+                      // Update the particle id of the particle in contact
```
Comment (blaisb) [inline source/dem/update_local_particle_containers.cc:90]: Is this comment still valid? I am not sure it makes sense no? (or it should apply to the line above no?)

**Review by blaisb — APPROVED** [review]
One last thing to check and it will be ok to merge

File: source/dem/update_local_particle_containers.cc (hunk 3 above)
Comment (acdaigneault) [inline source/dem/update_local_particle_containers.cc:90]:
```suggestion
```

**Pull request merged**